data/ipfs.go

```diff
@@ -257,7 +262,76 @@ func (i *IPFSHandle) Retrieve(ctx context.Context, path string, maxSize int64) (
 	}
 	r, ok := node.(files.File)
 	if !ok {
-		return nil, errors.New("received incorrect type from Unixfs().Get()")
+		return nil, fmt.Errorf("received incorrect type from Unixfs().Get()")
 	}
 	return io.ReadAll(r)
 }
+
+// PublishIPNSpath creates or updates an IPNS record with the content of a
+// filesystem path (a single file or a directory).
+//
+// The IPNS record is published under the scope of the private key identified
+// by the keyalias parameter. New keys can be created using method AddKeyToKeystore
+// and function NewIPFSkey() both available on this package.
+//
+// The execution of this method might take a while (some minutes),
+// so the caller must handle properly the logic by using goroutines, channels or other
+// mechanisms in order to not block the whole program execution.
+func (i *IPFSHandle) PublishIPNSpath(ctx context.Context, path string,
+	keyalias string) (coreiface.IpnsEntry, error) {
+	root, err := i.AddAndPin(ctx, path)
+	if err != nil {
+		return nil, err
+	}
+	c, err := cid.Parse(root)
+	if err != nil {
+		return nil, err
+	}
+	if keyalias == "" {
+		ck, err := i.CoreAPI.Key().Self(ctx)
+		if err != nil {
+			return nil, err
+		}
+		keyalias = ck.Name()
+	}
+	return i.CoreAPI.Name().Publish(
+		ctx,
+		ipath.IpfsPath(c),
+		options.Name.TTL(time.Duration(time.Minute*10)),
```

Comment:
this time.Duration conversion is unnecessary. I think you might spot more of these with https://github.com/mdempsky/unconvert